packages/library-base/src/commonMain/kotlin/io/realm/internal/RealmQueryImpl.kt

```diff
@@ -0,0 +1,255 @@
+    }
+
+    override fun sort(
+        propertyAndSortOrder: Pair<String, QuerySort>,
+        vararg additionalPropertiesAndOrders: Pair<String, QuerySort>
+    ): RealmQuery<E> {
+        val updatedDescriptors = (descriptors + QueryDescriptor.Sort(
+            propertyAndSortOrder,
+            *additionalPropertiesAndOrders
+        ))
+        return RealmQueryImpl(subQuery, updatedDescriptors, this)
+    }
+
+    override fun distinct(property: String, vararg extraProperties: String): RealmQuery<E> {
+        val updatedDescriptors =
+            (descriptors + QueryDescriptor.Distinct(property, *extraProperties))
+        return RealmQueryImpl(subQuery, updatedDescriptors, this)
+    }
+
+    override fun limit(limit: Int): RealmQuery<E> {
+        val updatedDescriptors = descriptors + QueryDescriptor.Limit(limit)
+        return RealmQueryImpl(subQuery, updatedDescriptors, this)
+    }
+
+    override fun first(): RealmSingleQuery<E> =
+        RealmSingleQueryImpl(realmReference, queryPointer, clazz, mediator)
+
+    override fun <T : Any> min(property: String, type: KClass<T>): RealmScalarQuery<T> =
+        GenericAggregatorQuery(
+            realmReference,
+            queryPointer,
+            mediator,
+            clazz,
+            property,
+            type,
+            AggregatorQueryType.MIN
+        )
+
+    override fun <T : Any> max(property: String, type: KClass<T>): RealmScalarQuery<T> =
+        GenericAggregatorQuery(
+            realmReference,
+            queryPointer,
+            mediator,
+            clazz,
+            property,
+            type,
+            AggregatorQueryType.MAX
+        )
+
+    override fun <T : Any> sum(property: String, type: KClass<T>): RealmScalarQuery<T> =
+        GenericAggregatorQuery(
+            realmReference,
+            queryPointer,
+            mediator,
+            clazz,
+            property,
+            type,
+            AggregatorQueryType.SUM
+        )
+
+//    override fun <T : Any> average(property: String, type: KClass<T>): RealmScalarQuery<T> =
+//        GenericAggregatorQuery(
+//            realmReference,
+//            queryPointer,
+//            mediator,
+//            clazz,
+//            property,
+//            type,
+//            AggregatorQueryType.AVERAGE
+//        )
+//
+//    override fun average(property: String): RealmScalarQuery<Double> =
+//        average(property, Double::class)
+
+    override fun count(): RealmScalarQuery<Long> =
+        CountQuery(realmReference, queryPointer, mediator)
+
+    override fun thaw(liveRealm: RealmReference): BaseResults<E> {
+        val liveResults = RealmInterop.realm_results_resolve_in(resultsPointer, liveRealm.dbPointer)
+        return ElementResults(liveRealm, liveResults, clazz, mediator)
+    }
+
+    override fun asFlow(): Flow<RealmResults<E>> = realmReference.owner
+        .registerObserver(this)
+        .onStart { realmReference.checkClosed() }
```

Comment:
Is this needed?

Reply:
I found the other flows do this check. I think this implementation might not respect the original intent since the check will be called as the first thing after calling `collect`, whereas in the other places where we do it is called right when calling `asFlow` and not as the first thing upon collection.